device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/MultiplexingClient.java

```diff
@@ -173,9 +213,61 @@ public void close() throws IOException
      * multiplexed connection.
      * <p>
      * Users should use {@link #registerDeviceClients(Iterable)} for registering multiple devices as it has some
-     * performance improvements over repeatedly calling this method. This method blocks on each registration, whereas
+     * performance improvements over repeatedly calling this method for individual device registrations. This method blocks on each registration, whereas
+     * {@link #registerDeviceClients(Iterable)} blocks on all of the registrations after starting them all asynchronously.
+     * <p>
+     * A device client can be unregistered using {@link #unregisterDeviceClient(DeviceClient)}, {@link #unregisterDeviceClient(DeviceClient, long)},
+     * {@link #unregisterDeviceClients(Iterable)}, or {@link #unregisterDeviceClients(Iterable, long)}. A device client will not be unregistered
+     * automatically if it encounters a non-retryable exception, so users are responsible for unregistering a device client
+     * when they no longer want it in this multiplexing client.
+     * <p>
+     * Up to {@link #MAX_MULTIPLEX_DEVICE_COUNT_AMQPS} devices can be registered on a multiplexed AMQPS connection,
+     * and up to {@link #MAX_MULTIPLEX_DEVICE_COUNT_AMQPS_WS} devices can be registered on a multiplexed AMQPS_WS connection.
+     * <p>
+     * If the multiplexing client is already open, then this device client will automatically
+     * be opened, too. If the multiplexing client is not open yet, then this device client will not be opened until
+     * {@link MultiplexingClient#open()} is called.
+     * <p>
+     * If the multiplexed connection is already open, then this call will add this device client to the
+     * multiplexed connection, and then will block until the registration has been completed.
```

Comment:
q - "and then will block until the registration has been completed" -> should this say "and then will block until the device client has been opened" instead?
also, adding an already opened device client to an opened mux connection will simply add it to the hashmap (or whichever collection is used) and will return immediately, correct?

Reply:
You can't register a device client that has already been opened, whether it was opened on a different multiplexing client, or just opened as a non-multiplexing client connection.

> should this say "and then will block until the device client has been opened"

I'm deliberately trying to use the phrase "register" rather than "open" in this context since the API itself is called "**register**DeviceClient"

Reply:
> You can't register a device client that has already been opened, whether it was opened on a different multiplexing client, or **just opened as a non-multiplexing client connection**.

Is this documented?